### ISSUE TITLE
Multiple "Your profile has been successfully updated." messages

### ISSUE THREAD

**Issue by ndebuhr** [hydroshare/hydroshare#3056]
If you go through a irods user delete+create cycle, multiple identical "Your profile has been successfully updated." messages will accumulate in the interface.

[011.zip](https://github.com/hydroshare/hydroshare/files/2683208/011.zip)

**Comment by martinseul** [comment]
I looked into this again for review and encountered an issue with deletion of an account: 
![image](https://user-images.githubusercontent.com/5427742/161101871-ec5a9706-0d59-490a-9acf-16a7fe7e8a6d.png)
@hyi could you please take quick look at the account

**Comment by eclark-cuahsi** [comment]
@martinseul to provide an update per 4/28 meeting

**Comment by martinseul** [comment]
@hyi so the deletion is not executed because the user has files in his account, this is not communicated to the user. Can we change the message that to either inform the user why the deletion was not executed or put up a warning that deletion of the account will also remove all related files and then delete diles and account. I prefer the second pathway to simplify deletion. It is common practice to remove files when deleting an account and the userzone is a temporary place for files anyway.

**Comment by devincowan** [comment]
Not sure if these are the same issues that @martinseul encountered... but making a not of them.
### Attempting account delete on beta

![image](https://user-images.githubusercontent.com/17934193/202484135-193a9805-5835-4e39-a197-845090ecfafa.png)

```
Nov 17 10:11:56 pid:28265 NOTICE: chlDelCollByAdmin delete access failure -819000
Nov 17 10:11:56 pid:28265 NOTICE: chlDelCollByAdmin cmlExecuteNoAnswerSql(rollback) succeeded
Nov 17 10:11:56 pid:28265 NOTICE: db_del_coll_by_admin_op cmlExecuteNoAnswerSql(rollback) succeeded
Nov 17 10:11:56 pid:28265 remote addresses: 152.54.3.110, ::1 ERROR: executeRuleAction Failed for msiDeleteUser status = -827000 CAT_INVALID_USER
Nov 17 10:11:56 pid:28265 NOTICE: executeRuleBody: Microservice or Action msiDeleteUser Failed with status -827000
Nov 17 10:11:56 pid:28265 NOTICE: execRuleNodeRes: applyRule Failed: acDeleteUserF1 with status -827000
Nov 17 10:11:56 pid:28265 remote addresses: 152.54.3.110, ::1 ERROR: executeRuleAction Failed for acDeleteUserF1 status = -827000 CAT_INVALID_USER
Nov 17 10:11:56 pid:28265 NOTICE: executeRuleBody: Microservice or Action acDeleteUserF1 Failed with status -827000
Nov 17 10:11:56 pid:28265 NOTICE: execRuleNodeRes: applyRule Failed: acDeleteUser with status -827000
Nov 17 10:11:56 pid:28265 DEBUG: 
Nov 17 10:11:56 pid:28265 remote addresses: 152.54.3.110, ::1 ERROR: applyRuleArg: -827000, [-]	/repos/irods/server/re/include/irods_re_plugin.hpp:326:irods::error irods::dynamic_operation_execution_manager<std::__1::tuple<>, RuleExecInfo *, irods::DONT_AUDIT_RULE>::call(std::string, std::string, OP, As &&...) [T = std::__1::tuple<>, C = RuleExecInfo *, Audit = irods::DONT_AUDIT_RULE, OP = std::__1::function<irods::error (const std::__1::basic_string<char> &, irods::re_pack_inp<std::__1::tuple<> > &, irods::unpack &&)>, As = <const std::__1::basic_string<char> &, irods::re_pack_inp<std::__1::tuple<> > &, irods::unpack>] :  status [CAT_INVALID_USER]  errno [] -- message [applyRuleUpdateParams failed for rule acDeleteUser]
	[-]	/repos/irods/plugins/rule_engines/irods_rule_engine_plugin-irods_rule_language/libirods_rule_engine_plugin-irods_rule_language.cpp:348:irods::error exec_rule(irods::default_re_ctx &, const std::string &, std::list<boost::any> &, irods::callback) :  status [CAT_INVALID_USER]  errno [] -- message [applyRuleUpdateParams failed for rule acDeleteUser]

Nov 17 10:11:56 pid:28265 NOTICE: rsGeneralAdmin: rcGeneralAdmin error -827000
```

So try to manually delete it from a shell:
```
$ iadmin rmuser dcowan
remote addresses: 152.54.3.110 ERROR: rcGeneralAdmin failed with error -827000 CAT_INVALID_USER 
Level 0: Invalid user
Level 1: DEBUG:
```

### Attempting account create on beta
![image](https://user-images.githubusercontent.com/17934193/202492089-2ee8a19b-545b-456a-b1f0-2712092c90c1.png)
```
Nov 17 10:40:17 pid:28920 NOTICE: execRuleNodeRes: applyRule Failed: acCreateUser with status -809000
Nov 17 10:40:17 pid:28920 DEBUG: 
Nov 17 10:40:17 pid:28920 remote addresses: 152.54.3.110, ::1 ERROR: applyRuleArg: -809000, [-]	/repos/irods/server/re/include/irods_re_plugin.hpp:326:irods::error irods::dynamic_operation_execution_manager<std::__1::tuple<>, RuleExecInfo *, irods::DONT_AUDIT_RULE>::call(std::string, std::string, OP, As &&...) [T = std::__1::tuple<>, C = RuleExecInfo *, Audit = irods::DONT_AUDIT_RULE, OP = std::__1::function<irods::error (const std::__1::basic_string<char> &, irods::re_pack_inp<std::__1::tuple<> > &, irods::unpack &&)>, As = <const std::__1::basic_string<char> &, irods::re_pack_inp<std::__1::tuple<> > &, irods::unpack>] :  status [CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME]  errno [] -- message [applyRuleUpdateParams failed for rule acCreateUser]
	[-]	/repos/irods/plugins/rule_engines/irods_rule_engine_plugin-irods_rule_language/libirods_rule_engine_plugin-irods_rule_language.cpp:348:irods::error exec_rule(irods::default_re_ctx &, const std::string &, std::list<boost::any> &, irods::callback) :  status [CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME]  errno [] -- message [applyRuleUpdateParams failed for rule acCreateUser]

Nov 17 10:40:17 pid:28920 NOTICE: rsGeneralAdmin: rcGeneralAdmin error -809000
Nov 17 10:40:17 pid:28923 remote addresses: 152.54.3.110, ::1 ERROR: [_rsGeneralAdmin:836] - [CAT_INVALID_USER: user does not exist

]
Nov 17 10:40:17 pid:28923 NOTICE: rsGeneralAdmin: rcGeneralAdmin error -827000
[dcowan@usersdev log]$ tail -f rodsLog.2022.11.16
Nov 17 10:40:17 pid:28920 NOTICE: execRuleNodeRes: applyRule Failed: acCreateUser with status -809000
Nov 17 10:40:17 pid:28920 DEBUG: 
Nov 17 10:40:17 pid:28920 remote addresses: 152.54.3.110, ::1 ERROR: applyRuleArg: -809000, [-]	/repos/irods/server/re/include/irods_re_plugin.hpp:326:irods::error irods::dynamic_operation_execution_manager<std::__1::tuple<>, RuleExecInfo *, irods::DONT_AUDIT_RULE>::call(std::string, std::string, OP, As &&...) [T = std::__1::tuple<>, C = RuleExecInfo *, Audit = irods::DONT_AUDIT_RULE, OP = std::__1::function<irods::error (const std::__1::basic_string<char> &, irods::re_pack_inp<std::__1::tuple<> > &, irods::unpack &&)>, As = <const std::__1::basic_string<char> &, irods::re_pack_inp<std::__1::tuple<> > &, irods::unpack>] :  status [CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME]  errno [] -- message [applyRuleUpdateParams failed for rule acCreateUser]
	[-]	/repos/irods/plugins/rule_engines/irods_rule_engine_plugin-irods_rule_language/libirods_rule_engine_plugin-irods_rule_language.cpp:348:irods::error exec_rule(irods::default_re_ctx &, const std::string &, std::list<boost::any> &, irods::callback) :  status [CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME]  errno [] -- message [applyRuleUpdateParams failed for rule acCreateUser]

Nov 17 10:40:17 pid:28920 NOTICE: rsGeneralAdmin: rcGeneralAdmin error -809000
Nov 17 10:40:17 pid:28923 remote addresses: 152.54.3.110, ::1 ERROR: [_rsGeneralAdmin:836] - [CAT_INVALID_USER: user does not exist

]
Nov 17 10:40:17 pid:28923 NOTICE: rsGeneralAdmin: rcGeneralAdmin error -827000
Nov 17 10:41:27 pid:28982 NOTICE: bindVar[1]=/hydroshareuserdevZone/home
Nov 17 10:41:27 pid:28982 NOTICE: bindVar[2]=/hydroshareuserdevZone/home/dcowan
Nov 17 10:41:27 pid:28982 NOTICE: bindVar[3]=dcowan
Nov 17 10:41:27 pid:28982 NOTICE: bindVar[4]=hydroshareuserdevZone
Nov 17 10:41:27 pid:28982 NOTICE: bindVar[5]=
Nov 17 10:41:27 pid:28982 NOTICE: bindVar[6]=
Nov 17 10:41:27 pid:28982 NOTICE: bindVar[7]=
Nov 17 10:41:27 pid:28982 NOTICE: _cllExecSqlNoResult: SQLExecDirect error: -1 sql:insert into R_COLL_MAIN (coll_id, parent_coll_name, coll_name, coll_owner_name, coll_owner_zone, coll_type, coll_info1, coll_info2, create_ts, modify_ts) values (nextval('R_ObjectID'), ?, ?, ?, ?, ?, ?, ?, ?, ?)
Nov 17 10:41:27 pid:28982 NOTICE: SQLSTATE: S1010
Nov 17 10:41:27 pid:28982 NOTICE: SQLCODE: 0
Nov 17 10:41:27 pid:28982 NOTICE: SQL Error message: [unixODBC][Driver Manager]Function sequence error
Nov 17 10:41:27 pid:28982 NOTICE: SQLSTATE: 23505
Nov 17 10:41:27 pid:28982 NOTICE: SQLCODE: 1
Nov 17 10:41:27 pid:28982 NOTICE: SQL Error message: ERROR: duplicate key value violates unique constraint "idx_coll_main3";
Error while executing the query
Nov 17 10:41:27 pid:28982 NOTICE: chlRegCollByAdmin cmlExecuteNoAnswerSQL(insert) failure -809000
Nov 17 10:41:27 pid:28982 NOTICE: chlRegCollByAdmin cmlExecuteNoAnswerSql(rollback) succeeded
Nov 17 10:41:27 pid:28982 remote addresses: 152.54.3.110, ::1 ERROR: executeRuleAction Failed for msiCreateCollByAdmin status = -809000 CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME
Nov 17 10:41:27 pid:28982 NOTICE: executeRuleBody: Microservice or Action msiCreateCollByAdmin Failed with status -809000
Nov 17 10:41:27 pid:28982 NOTICE: execRuleNodeRes: applyRule Failed: acCreateCollByAdmin with status -809000
Nov 17 10:41:27 pid:28982 remote addresses: 152.54.3.110, ::1 ERROR: executeRuleAction Failed for acCreateCollByAdmin status = -809000 CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME
Nov 17 10:41:27 pid:28982 NOTICE: executeRuleBody: Microservice or Action acCreateCollByAdmin Failed with status -809000
Nov 17 10:41:27 pid:28982 NOTICE: execRuleNodeRes: applyRule Failed: acCreateUserZoneCollections with status -809000
Nov 17 10:41:27 pid:28982 remote addresses: 152.54.3.110, ::1 ERROR: executeRuleAction Failed for acCreateUserZoneCollections status = -809000 CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME
Nov 17 10:41:27 pid:28982 NOTICE: executeRuleBody: Microservice or Action acCreateUserZoneCollections Failed with status -809000
Nov 17 10:41:27 pid:28982 NOTICE: execRuleNodeRes: applyRule Failed: acCreateDefaultCollections with status -809000
Nov 17 10:41:27 pid:28982 remote addresses: 152.54.3.110, ::1 ERROR: executeRuleAction Failed for acCreateDefaultCollections status = -809000 CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME
Nov 17 10:41:27 pid:28982 NOTICE: executeRuleBody: Microservice or Action acCreateDefaultCollections Failed with status -809000
Nov 17 10:41:27 pid:28982 NOTICE: execRuleNodeRes: applyRule Failed: acCreateUserF1 with status -809000
Nov 17 10:41:27 pid:28982 remote addresses: 152.54.3.110, ::1 ERROR: executeRuleAction Failed for acCreateUserF1 status = -809000 CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME
Nov 17 10:41:27 pid:28982 NOTICE: executeRuleBody: Microservice or Action acCreateUserF1 Failed with status -809000
Nov 17 10:41:27 pid:28982 NOTICE: execRuleNodeRes: applyRule Failed: acCreateUser with status -809000
Nov 17 10:41:27 pid:28982 DEBUG: 
Nov 17 10:41:27 pid:28982 remote addresses: 152.54.3.110, ::1 ERROR: applyRuleArg: -809000, [-]	/repos/irods/server/re/include/irods_re_plugin.hpp:326:irods::error irods::dynamic_operation_execution_manager<std::__1::tuple<>, RuleExecInfo *, irods::DONT_AUDIT_RULE>::call(std::string, std::string, OP, As &&...) [T = std::__1::tuple<>, C = RuleExecInfo *, Audit = irods::DONT_AUDIT_RULE, OP = std::__1::function<irods::error (const std::__1::basic_string<char> &, irods::re_pack_inp<std::__1::tuple<> > &, irods::unpack &&)>, As = <const std::__1::basic_string<char> &, irods::re_pack_inp<std::__1::tuple<> > &, irods::unpack>] :  status [CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME]  errno [] -- message [applyRuleUpdateParams failed for rule acCreateUser]
	[-]	/repos/irods/plugins/rule_engines/irods_rule_engine_plugin-irods_rule_language/libirods_rule_engine_plugin-irods_rule_language.cpp:348:irods::error exec_rule(irods::default_re_ctx &, const std::string &, std::list<boost::any> &, irods::callback) :  status [CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME]  errno [] -- message [applyRuleUpdateParams failed for rule acCreateUser]

Nov 17 10:41:27 pid:28982 NOTICE: rsGeneralAdmin: rcGeneralAdmin error -809000
Nov 17 10:41:27 pid:28985 remote addresses: 152.54.3.110, ::1 ERROR: [_rsGeneralAdmin:836] - [CAT_INVALID_USER: user does not exist

]
Nov 17 10:41:27 pid:28985 NOTICE: rsGeneralAdmin: rcGeneralAdmin error -827000
```

Attempt manually create in shell:
```
iadmin mkuser dcowan rodsuser
remote addresses: 152.54.3.110 ERROR: rcGeneralAdmin failed with error -809000 CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME 
Level 0: Error -809000 CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME
Level 1: DEBUG: 
```

**Comment by devincowan** [comment]
Seems like my inability do remove/add iRods account is perhaps related to https://github.com/irods/irods/issues/3931.
Or perhaps https://github.com/irods/irods/issues/3798 or https://github.com/irods/irods/issues/1976

**Comment by devincowan** [comment]
Looks like this issue (#3056) is resolved as of HS 2.6.1
So I created a new issue https://github.com/hydroshare/hydroshare/issues/5104 for the broken IRODS account deletion
